packages/stable/src/api/documents/types.gen.ts

```diff
@@ -356,8 +356,38 @@ export type DocumentsAggregateCountRequest = {
 export interface DocumentsAggregateCountResponse {
     items: DocumentsAggregateCountItem[];
 }
-export type DocumentsAggregateRequest = DocumentsAggregateCountRequest | DocumentsAggregateUniqueValuesRequest | DocumentsAggregateAllUniqueValuesRequest;
-export type DocumentsAggregateResponse = DocumentsAggregateCountResponse | DocumentsAggregateUniqueValuesResponse | DocumentsAggregateAllUniqueValuesResponse;
+export type DocumentsAggregateRequest = DocumentsAggregateCountRequest | DocumentsAggregateUniqueValuesRequest | DocumentsAggregateUniquePropertiesRequest | DocumentsAggregateAllUniqueValuesRequest;
+export type DocumentsAggregateResponse = DocumentsAggregateCountResponse | DocumentsAggregateUniqueValuesResponse | DocumentsAggregateUniquePropertiesResponse | DocumentsAggregateAllUniqueValuesResponse;
+export interface DocumentsAggregateUniquePropertiesItem {
+    /**
+     * Number of properties with this name
+     * @format int64
+     */
+    count: number;
+    /** A property name */
+    values: string[];
+}
+/**
+ * Find all metadata property names
+ */
+export type DocumentsAggregateUniquePropertiesRequest = {
+    search?: {
+        query: string;
+    };
+} & DocumentSearchFilter & {
+    aggregate?: "uniqueProperties";
+    properties?: {
```

Comment:
Interesting. Seems there's a codegen bug as these fields are mandatory.

Reply:
This is fine though. Adjusting this to required fields later won't cause a breaking change, so I'm clicking approve.

Reply:
The object this is based on is quite complex, with a big `allOf` with multiple inline objects. I am not surprised it is difficult for the autogenerator to deal with.

Perhaps it would work better if the `required` clause in the spec was inside the same object where the required properties are. I'll test that

Reply:
I have updated the service contract here: https://github.com/cognitedata/service-contracts/pull/2020